source/develop/release-management/features/sla/hosted-engine-new-deployment.html.md

```diff
@@ -0,0 +1,152 @@
+
+To run the script user shoud install ansible and run:   
+    ansible-playbook hosted-engine.yml   
+The script will install all the packages and will start the deployment.   
+The vars file needs to be populated manually for now. 
+
+## Issues
+
+1. Host installation fails during network setup stage  
+   The reason for this is that vdsm setup removes the routing table.   
+   Workaround: rewrite the routing table after the host gets to the "non operational" state and then save the network configuration 
+   using vdsm-client. This causes the host to go to "up" state almost immediatly.   
+   Permanent solution: handle this in host deploy code.   
+2. Adding hosted engine disk takes a long time   
+   This might not be a problem if using a cow format and not raw. But this wasn't tested yet.   
+3. When using fqdn that resolves to a real ip the bootstrap vm tries to access the real ip and the installation fails 
+   Workaround: add a line to /etc/hosts on the host with the fqdn and the vm temporary ip.  
+               add a line to /etc/hosts on the vm with localhost and fqdn - not implemented.   
+   After successfull installation those lines need to be removed.   
+4. When hosted engine with iscsi domain the storage domain creation fails  
+   This seems like a bug in Ansible ovirt_storage_domain module which doesn't support this feature: 
+   https://www.ovirt.org/develop/release-management/features/storage/discard-after-delete/  
+   Until this is fixed we can't deploy with iscsi.  
+   Workaround: use direct api command until the Ansible module is fixed - not implemented.   
+5. Gluster and fibre channel storage  
+   Not implemented   
+6. When creating the configuration archive agent fails to read it  
+   Ansible's archive module creates a compressed tar archive, the agent expects a non compressed one.   
+   Workaround: create the archive using the command module.    
+7. After installing vdsm libvirt commands need to be authenticted with vdsm user  
+   Workaround: configure libvirt to listen to tls and preform the commands using: qemu+tls:///{IP}/system
+8. The fheanswers file in the configuration archive should have the answers the user submitted   
+   Not implemented
+
+## Detailed flow
+
+1. Install needed pacakges: libvirt, ovirt-engine-appliance, virt-install and python-ovirt-engine-sdk4
+2. Create a directory for the bootstrap vm 
+3. Extract the appliance to the local directory
+4. Create cloud init files that will run engine setup, and generate an iso cd
+5. Create the bootstrap vm, cloud init will install the engine
+6. Add the fqdn and the bootstrap vm ip to /etc/hosts on the host
+7. Wait for the enigne setup to complete
+8. Add the fqdn and the bootstrap vm ip to /etc/hosts on the vm 
+9. Add host 
+10. Wait for host to become non operational
+11. Restore the routing table on the host
+12. Save the network configuration 
+13. Wait for host to become up
```

Comment:
Not clear

Reply:
The setup configures network and waits for the host to get an "up" status.